mvc/controllers/areas.rst

```diff
@@ -1,8 +1,93 @@
-.. include:: /../common/stub-topic.txt
+Areas
+=====
+By `Tom Archer`
 
-|stub-icon| Areas
-===========================================
+:doc:`Areas </controllers/areas>` provide a way to separate a large MVC application into semantically-related groups of models, views, and controllers. Let's take a look at an example to illustrate how Areas are created and used. Let's say you have a store app that has two distinct groupings of controllers and views: Products and Services.
 
-.. include:: /../common/stub-notice.txt
+Instead of having all of the controllers housed under the Controllers parent directory, and all the views housed under the Views parent directory, you could use Areas to group your views and controllers according to the area (or logical grouping) with which they're associated.
 
-.. _issue: https://github.com/aspnet/Docs/issues/121
+- Project name
+
+  - Areas
+
+    - Products
+
+      - Controllers
+
+        - HomeController
+
+      - Views
+
+        - Home
+
+          - Index.cshtml
+
+    - Services
+
+      - Controllers
+
+        - HomeController
+
+      - Views
+
+        - Home
+
+          - Index.cshtml
+
+Once you've defined the folder hierarchy, you need to tell MVC that each controller is associated with an area. You do that by decorating the controller name with the ``[Area]`` attribute.
+
+.. code-block:: c#
+  :emphasize-lines: 4
+
+  ...
+  namespace MyStore.Areas.Products.Controllers
+  {
+      [Area("Products")]
+      public class HomeController : Controller
+      {
+          // GET: /<controller>/
+          public IActionResult Index()
+          {
+              return View();
+          }
+      }
+  }
+
+The final step is to set up a route definition that works with your newly created areas. The :doc:`routing` article goes into detail about how to create route definitions, including using conventional routes versus attribute routes. In this example, we'll use a conventional route. To do so, simply open the ``startup.cs`` file and modify it by adding the highlighted route definition below.
+
+.. code-block:: c#
+  :emphasize-lines: 4-6
+
+  ...
+  app.UseMvc(routes =>
+  {
+    routes.MapRoute(name: "areaRoute",
+      template: "{area:exists}/{controller=Home}/{action=Index}");
+
+    routes.MapRoute(
+        name: "default",
+        template: "{controller=Home}/{action=Index}");
+  });
+
+Now, when the user browses to *http://<yourApp>/products*, the ``Index`` action method of the ``HomeController`` in the ``Products`` area will be invoked.
+
+Linking between areas
+---------------------
+
+To link between areas, you simply specify the area in which the controller is defined. If the controller doesn't exist within an area, use an empty string.
```

Comment:
Maybe say "If the controller is not part of an area"? ("Doesn't exist..." makes it sound like an error condition...)